packages/grant-explorer/src/features/collection/AllCollections.tsx

```diff
@@ -0,0 +1,221 @@
+  collection,
+  isActiveRound,
+  backToLists,
+}: CollectionPreviewProps) => {
+  const [, copy] = useCopyToClipboard();
+
+  const { chainId, roundId } = useParams();
+  const round = useRoundById(chainId as string, roundId as string);
+  const [, setShowCartNotification] = useState(false);
+  const [, setCurrentProjectAddedToCart] = useState<Project>({} as Project);
+
+  return (
+    <div>
+      <div className="flex my-4 justify-between">
+        <nav className="flex" aria-label="Breadcrumb">
+          <ol className="inline-flex items-center space-x-1 md:space-x-3">
+            <li className="inline-flex items-center">
+              <span role="button" onClick={backToLists}>
+                My Lists
+              </span>
+            </li>
+            <li aria-current="page">
+              <div className="flex items-center">
+                <ChevronRightIcon className="w-3 h-3" strokeWidth={3} />
+                <span className="ml-1 font-medium md:ml-2">
+                  {collection.title} ({collection.projects.length})
+                </span>
+              </div>
+            </li>
+          </ol>
+        </nav>
+
+        <Button
+          $variant="secondary"
+          className="flex flex-nowrap items-center"
+          onClick={() =>
+            copy(
+              `${window.location.protocol}//${window.location.host}/#/collection?data=${collection.enc}`
+            )
+          }
+        >
+          <LinkIcon className="w-4 h-4 mr-2" strokeWidth={3} />
+          <span>Share list</span>
+        </Button>
+      </div>
+      <ProjectList
+        projects={collection.projects}
+        roundRoutePath={`/round/${chainId}/${roundId}`}
+        isBeforeRoundEndDate={isActiveRound}
+        isInCollection={true}
+        roundId={roundId as string}
+        chainId={chainId as unknown as ChainId}
+        round={round.round!}
+        canUseCollections={true}
+        setShowCartNotification={setShowCartNotification}
+        setCurrentProjectAddedToCart={setCurrentProjectAddedToCart}
+      />
+    </div>
+  );
+};
+
+const AllCollectionsView = ({ projects, isActiveRound }: Props) => {
+  const [collections, setCollections] = useState<Collection[]>([]);
+  const [loadingCollections, setLoadingCollections] = useState<boolean>(false);
+  const [activeCollection, setActiveCollection] = useState<ActiveCollection>();
+  const { chainId, roundId } = useParams();
+
+  const loadAllCollections = useCallback(async (round: string) => {
+    setLoadingCollections(true);
+    const _collections = await getAllCollections(round);
+    setCollections(
+      _collections.map((i) => ({
+        ...i,
+        enc: encodeURIComponent(encodeCollection(JSON.stringify(i))),
+      }))
+    );
+    setLoadingCollections(false);
+  }, []);
+
+  useEffect(() => {
+    // load all the collections for this round here
+    loadAllCollections(`${chainId}:${roundId}`);
+  }, [chainId, roundId, projects, loadAllCollections]);
```

Comment:
We can simplify a lot of this with useSWR. That way we don't need to use `useState`, `useEffect`, and `useCallback`. We also get loading and errors states.